### DISCUSSION TITLE
Filter by Mood

### DISCUSSION THREAD

**Discussion by cbjones93** [nss-day-cohort-47/C47-Discussions#46] [Help Me] [answered]
Having difficulty filtering by mood, currently getting in console.log:
User wants to see posts with a mood of Okay
main.js:51 Uncaught ReferenceError: singlePost is not defined

Just not sure where to define the singePost. 
Can't seem to find an example on google that's not a filter by number.
```
applicationElement.addEventListener("change", event => {
	if (event.target.id === "mood"){
		const moodValue = (event.target.value)
		console.log(`User wants to see posts with a mood of ${moodValue}`)
		showFilteredMoodPosts(moodValue);
	}
})

const showFilteredMoodPosts = () =>{
	const filteredMood = usePostCollection().filter(singePost=>{
		if(singlePost.mood === "Okay"){
			return singePost
		}
	})
	const postMoodElement = document.querySelector(".journalEntryBox");
	postMoodElement.innerHTML = journalList(filteredMood);
}
```
link to my repo:https:https://github.com/cbjones93/Daily-Journal/pull/4

**Comment by BryanNilsen** [comment]
looks like you have a typo.
in your filter, you are calling the item "**singePost**" whereas you're trying to check the mood property of "**singlePost**" in your conditional.

**Reply by cbjones93** [reply]
Well, that certainly got rid of the error! Typos are awful, thanks for pointing it out. Still not having luck getting the filter to work.

**Comment by cbjones93** [comment, marked as answer]
Turns out there were even more typos! Gotta be more careful when writing my code. 
Also turns out I've been using my create new post form as my event listener, so I need to make sure to make a new selection box for this but with Bryans help I was able to get it to work.  I was able to plug in moodValue in function showFilteredMoodPosts to reuse the function for all the different values. Here is my code:
```applicationElement.addEventListener("change", event => {
	if (event.target.id === "mood") {
		const moodValue = (event.target.value)
		console.log(`User wants to see posts with a mood of ${moodValue}`)
		showFilteredMoodPosts(moodValue);
	}
})
const showFilteredMoodPosts = (moodValue) => {
	const filteredMood = usePostCollection().filter(singlePost => {
		if (singlePost.mood === moodValue) {
			return singlePost
		}
	})
	const postMoodElement = document.querySelector(".journalEntryBox");
	postMoodElement.innerHTML = journalList(filteredMood);
}
```
